api/pkg/apis/v1alpha1/managers/campaigncontainers/campaign-container-manager.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package campaigncontainers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/model"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/contexts"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/managers"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers/states"
+	"github.com/eclipse-symphony/symphony/coa/pkg/logger"
+
+	observability "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/observability"
+	observ_utils "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/observability/utils"
+)
+
+var log = logger.NewLogger("coa.runtime")
+
+type CampaignContainersManager struct {
+	managers.Manager
+	StateProvider states.IStateProvider
+}
+
+func (s *CampaignContainersManager) Init(context *contexts.VendorContext, config managers.ManagerConfig, providers map[string]providers.IProvider) error {
+	err := s.Manager.Init(context, config, providers)
+	if err != nil {
+		return err
+	}
+	stateprovider, err := managers.GetStateProvider(config, providers)
+	if err == nil {
+		s.StateProvider = stateprovider
+	} else {
+		return err
+	}
+	return nil
+}
+
+func (t *CampaignContainersManager) DeleteState(ctx context.Context, name string, namespace string) error {
+	ctx, span := observability.StartSpan("CampaignContainersManager", ctx, &map[string]string{
+		"method": "DeleteState",
+	})
+	var err error = nil
+	defer observ_utils.CloseSpanWithError(span, &err)
+
+	err = t.StateProvider.Delete(ctx, states.DeleteRequest{
+		ID: name,
+		Metadata: map[string]interface{}{
+			"namespace": namespace,
+			"group":     model.SolutionGroup,
```

Comment:
Wrong group

Reply:
Good catch. campaign is the only resource which is not supported in materialize.